### PR TITLE
fix: handle community shard unassignment and update

### PR DESCRIPTION
**Pull request by richard-ramos**
Should fix the issue described in https://github.com/status-im/status-desktop/issues/13267

### CODE REVIEW

File: wakuv2/waku.go [hunk 1]
```diff
@@ -610,6 +610,24 @@ func (w *Waku) getPubsubTopic(topic string) string {
...
+		return errors.New("only available for full nodes")
+	}
+
+	topic = w.getPubsubTopic(topic)
+
+	if !w.node.Relay().IsSubscribed(topic) {
+		return nil
+	}
+
+	w.node.Relay().RemoveTopicValidator(topic)
```
Comment (chaitanyaprem) [inline wakuv2/waku.go]: This is already done as part of w.node.Relay().Unsubscribe. 
No need to do this specifically again.

File: wakuv2/waku.go [hunk 2]
```diff
@@ -610,6 +610,24 @@ func (w *Waku) getPubsubTopic(topic string) string {
 	return topic
 }
 
+func (w *Waku) unsubscribeFromPubsubTopicWithWakuRelay(topic string) error {
+	if w.settings.LightClient {
```
Comment (chaitanyaprem) [inline wakuv2/waku.go]: This check is also done in line 1559 below already, hence is duplicate and can be removed.

**Review by chaitanyaprem — COMMENTED** [review]
Reviewed only parts that are invoking Waku API. 

Also, wondering if it is a good idea to just allow a community to switch a shard without some sort of migration warning. But this should be done at UI level i guess. 
Also, it may be better to indicate the impact of doing this, as message history would be lost and any other such impact of changing shard.

File: protocol/messenger_communities.go [hunk 3]
```diff
@@ -715,6 +715,16 @@ func (m *Messenger) subscribeToCommunityShard(communityID []byte, shard *shard.S
 	return m.transport.SubscribeToPubsubTopic(pubsubTopic, pubK)
 }
 
+func (m *Messenger) unsubscribeFromShard(shard *shard.Shard) error {
+	if m.transport.WakuVersion() != 2 {
+		return nil
+	}
```
Comment (igor-sirotin) [inline protocol/messenger_communities.go:721]: This check is redundant, as it's already done it `transport.UnsubscribeFromPubsubTopic`

File: protocol/messenger_communities.go [hunk 4]
```diff
@@ -2829,21 +2875,39 @@ func (m *Messenger) HandleCommunityShardKey(state *ReceivedMessageState, message
 	return nil
 }
 
-func (m *Messenger) handleCommunityShardAndFiltersFromProto(community *communities.Community, shard *shard.Shard, privateKeyBytes []byte) error {
+func (m *Messenger) handleCommunityShardAndFiltersFromProto(community *communities.Community, clock uint64, shard *shard.Shard, privateKeyBytes []byte) error {
```
Comment (igor-sirotin) [inline protocol/messenger_communities.go]: What's the `clock` is needed here for? I don't see it being used.

File: wakuv2/waku.go (hunk 2 above)
Comment (igor-sirotin) [reply wakuv2/waku.go]: But this also prevents of calling `RemoveTopicValidator`. Not sure what it does, just noting.

**Review by igor-sirotin — APPROVED** [review]

**Pull request merged**